src/core/composer/qgsatlascomposition.cpp

```diff
@@ -436,9 +436,13 @@ bool QgsAtlasComposition::prepareForFeature( const int featureI, const bool upda
 void QgsAtlasComposition::computeExtent( QgsComposerMap *map )
 {
   // QgsGeometry::boundingBox is expressed in the geometry"s native CRS
-  // We have to transform the grometry to the destination CRS and ask for the bounding box
+  // We have to transform the geometry to the destination CRS and ask for the bounding box
   // Note: we cannot directly take the transformation of the bounding box, since transformations are not linear
-  mTransformedFeatureBounds = currentGeometry( map->crs() ).boundingBox();
+  QgsGeometry g( currentGeometry( map->crs() ) );
+  // Rotating the geometry, so the bounding box is correct wrt map rotation
+  if ( map->mapRotation() != 0.0 )
+    g.rotate( map->mapRotation(), g.centroid().asPoint() );
```

Comment:
Is using the centroid of the geometry correct? I'm thinking possibly this should be the center of the geometry's original bounding box instead.